app/services/unwithdraw_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class UnwithdrawService
+  def call(edition, user = nil)
+    Document.transaction(requires_new: true) do
+      raise "edition must be withdrawn to be unwithdrawn" unless edition.withdrawn?
+
+      published_status = edition.status.details.published_status
```

Comment:
Same as the previous comment - it's a little vague that the current status is for a withdrawal, due to the separation of `edition.withdrawn?` and `edition.status`.

Reply:
Yep i agree, i've updated to use a withdrawal variable.